flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCTableSourceSinkFactory.java

```diff
@@ -0,0 +1,236 @@
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.apache.flink.table.descriptors.ConnectorDescriptorValidator.CONNECTOR_PROPERTY_VERSION;
+import static org.apache.flink.table.descriptors.ConnectorDescriptorValidator.CONNECTOR_TYPE;
+import static org.apache.flink.table.descriptors.Schema.SCHEMA;
+import static org.apache.flink.table.descriptors.Schema.SCHEMA_FROM;
+import static org.apache.flink.table.descriptors.Schema.SCHEMA_NAME;
+import static org.apache.flink.table.descriptors.Schema.SCHEMA_TYPE;
+
+/**
+ * Factory for creating configured instances of {@link JDBCTableSource} and {@link JDBCUpsertTableSink}.
+ */
+public class JDBCTableSourceSinkFactory implements
+	StreamTableSourceFactory<Row>,
+	StreamTableSinkFactory<Tuple2<Boolean, Row>> {
+
+	public static final String CONNECTOR_DRIVER_NAME = "connector.driver.name";
+	public static final String CONNECTOR_URL = "connector.url";
+	public static final String CONNECTOR_TABLE_NAME = "connector.table.name";
+	public static final String CONNECTOR_USERNAME = "connector.username";
+	public static final String CONNECTOR_PASSWORD = "connector.password";
+
+	public static final String CONNECTOR_SCAN_PARTITION_COLUMN_NAME = "connector.scan.partition.column.name";
+	public static final String CONNECTOR_SCAN_PARTITION_SIZE = "connector.scan.partition.size";
+	public static final String CONNECTOR_SCAN_PARTITION_LOWER = "connector.scan.partition.lower";
+	public static final String CONNECTOR_SCAN_PARTITION_UPPER = "connector.scan.partition.upper";
+
+	public static final String CONNECTOR_LOOKUP_CACHE_MAX_SIZE = "connector.lookup.cache.max-size";
+	public static final String CONNECTOR_LOOKUP_CACHE_EXPIRE_MS = "connector.lookup.cache.expire-ms";
+	public static final String CONNECTOR_LOOKUP_MAX_RETRY_TIMES = "connector.lookup.max-retry-times";
+
+	public static final String CONNECTOR_SINK_UPDATE_MODE = "connector.sink.update-mode";
+	public static final String CONNECTOR_SINK_FLUSH_MAX_SIZE = "connector.sink.flush.max-size";
+	public static final String CONNECTOR_SINK_FLUSH_INTERVAL_MS = "connector.sink.flush.interval-ms";
+	public static final String CONNECTOR_SINK_MAX_RETRY_TIMES = "connector.sink.max-retry-times";
+
+	@Override
+	public Map<String, String> requiredContext() {
+		Map<String, String> context = new HashMap<>();
+		context.put(CONNECTOR_TYPE, "jdbc"); // jdbc
+		context.put(CONNECTOR_PROPERTY_VERSION, "1"); // backwards compatibility
+		return context;
+	}
+
+	@Override
+	public List<String> supportedProperties() {
+		List<String> properties = new ArrayList<>();
+
+		// common options
+		properties.add(CONNECTOR_DRIVER_NAME);
+		properties.add(CONNECTOR_URL);
+		properties.add(CONNECTOR_TABLE_NAME);
+		properties.add(CONNECTOR_USERNAME);
+		properties.add(CONNECTOR_PASSWORD);
+
+		// scan partition options
+		properties.add(CONNECTOR_SCAN_PARTITION_COLUMN_NAME);
```

Comment:
A few concern about this scan partition api.
What about hive and spark?

Reply:
Spark also have partition api. See [https://spark.apache.org/docs/latest/sql-data-sources-jdbc.html].